src/tcache.c

```diff
@@ -703,35 +702,44 @@ tcache_init(tsd_t *tsd, tcache_slow_t *tcache_slow, tcache_t *tcache,
 	}
 	/*
 	 * For small size classes beyond tcache_max(i.e.
-	 * tcache_nhbins< NBINS), their cache bins are initialized to a state
+	 * tcache_nbins< NBINS), their cache bins are initialized to a state
 	 * to safely and efficiently fail all fastpath alloc / free, so that
-	 * no additional check around tcache_nhbins is needed on fastpath.
+	 * no additional check around tcache_nbins is needed on fastpath.
+	 * Yet we still store the default ncached_max generated by
+	 * tcache_ncached_max_compute in the bin_info for future usage.
 	 */
-	for (unsigned i = tcache_nhbins; i < SC_NBINS; i++) {
-		/* Disabled small bins. */
+	for (unsigned i = tcache_nbins; i < TCACHE_NBINS_MAX; i++) {
 		cache_bin_t *cache_bin = &tcache->bins[i];
-		void *fake_stack = mem;
-		size_t fake_offset = 0;
 
-		cache_bin_init(cache_bin, &tcache_bin_info[i], fake_stack,
-		    &fake_offset);
-		assert(tcache_small_bin_disabled(i, cache_bin));
+		if (i < SC_NBINS) {
+			/* Disabled small bins. */
+			void *fake_stack = mem;
+			size_t fake_offset = 0;
+			cache_bin_info_t fake_info;
+			cache_bin_info_init(&fake_info, 0);
+
+			cache_bin_init(cache_bin, &fake_info, fake_stack,
+			    &fake_offset);
+			assert(tcache_small_bin_disabled(i, cache_bin));
+		}
+
+		cache_bin_info_init(&cache_bin->bin_info,
```

Comment:
This needs to be in an `else` branch, otherwise the small bins are done twice.

How much convenience would it bring, if we also init disabled large bins with fake stacks like the small ones? I was opposed to it before but feel we should be open to it, if the benefit is meaningful (in other words, it's up to you if making that change).

Reply:
I don't see many benefits in doing so unless we want to change how fast path works. It could be done but that sounds like another PR.